src/sqlfluff/cli/commands.py

```diff
@@ -551,19 +553,31 @@ def fix(force, paths, processes, bench=False, fixed_suffix="", logger=None, **kw
                     _completion_message(config)
             elif c == "n":
                 click.echo("Aborting...")
+                exit_code = 1
             else:  # pragma: no cover
                 click.echo("Invalid input, please enter 'Y' or 'N'")
                 click.echo("Aborting...")
+                exit_code = 1
     else:
         click.echo("==== no fixable linting violations found ====")
-        if result.num_violations(types=SQLLintError, fixable=False) > 0:
-            click.echo(  # pragma: no cover
-                "  [{} unfixable linting violations found]".format(
-                    result.num_violations(types=SQLLintError, fixable=False)
-                )
-            )
         _completion_message(config)
 
+    if result.num_violations(types=SQLLintError, fixable=False) > 0:
+        click.echo(
+            "  [{} unfixable linting violations found]".format(
+                result.num_violations(types=SQLLintError, fixable=False)
+            )
+        )
+        exit_code = 1
```

Comment:
Again I don't like that this is hidden if there are fixable errors so have moved it out of the else statement